### PR TITLE
ARC-1294 adding change to make any graphql error retry without 'changedFiles'

### PR DESCRIPTION
**Pull request by mboudreau**
Testing out a potential solution for PCS-88578.  We might not be catching an error correctly because the graphql api is problematic.

### CODE REVIEW

**Review by bgvozdev — DISMISSED** [review]

File: src/github/client/github-client-errors.ts [hunk 1]
```diff
@@ -68,10 +70,13 @@ export class GithubClientGraphQLError extends GithubClientError {
...
-
 }
 
-export const isChangedFilesError = (err: GithubClientGraphQLError): boolean =>
-	!!err?.errors?.find(e => e.message?.includes("changedFiles"));
+const logger = getLogger("github.errors");
+export const isChangedFilesError = (err: GithubClientGraphQLError | GithubClientError): boolean => {
+	logger.warn({error: err}, "isChangedFilesError");
+	return err instanceof GithubClientGraphQLError || !(err instanceof RateLimitingError || err instanceof GithubClientTimeoutError);
+	// return !!err?.errors?.find(e => e.message?.includes("changedFiles"));
```
Comment (joshkay10) [inline src/github/client/github-client-errors.ts:79]: /nit delete comment 🫥

File: src/github/client/github-client-errors.ts [hunk 2]
```diff
@@ -68,10 +70,13 @@ export class GithubClientGraphQLError extends GithubClientError {
...
 		this.errors = errors;
 	}
-
 }
 
-export const isChangedFilesError = (err: GithubClientGraphQLError): boolean =>
-	!!err?.errors?.find(e => e.message?.includes("changedFiles"));
+const logger = getLogger("github.errors");
+export const isChangedFilesError = (err: GithubClientGraphQLError | GithubClientError): boolean => {
+	logger.warn({error: err}, "isChangedFilesError");
```
Comment (joshkay10) [inline src/github/client/github-client-errors.ts:77]: Update the warning message to the new checks

**Review by joshkay10 — DISMISSED** [review]

File: src/github/client/github-client-errors.ts (hunk 1 above)
Comment (mboudreau) [reply src/github/client/github-client-errors.ts:79]: this is just a test PR, will clean this up later.

File: src/github/client/github-client-errors.ts (hunk 2 above)
Comment (mboudreau) [reply src/github/client/github-client-errors.ts:77]: not a bad idea.

**Review by joshkay10 — APPROVED** [review]

**Pull request merged**